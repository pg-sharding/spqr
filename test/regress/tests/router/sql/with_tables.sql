\c spqr-console

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;

ALTER DISTRIBUTION ds1 ATTACH RELATION orders DISTRIBUTION KEY id;
ALTER DISTRIBUTION ds1 ATTACH RELATION delivery DISTRIBUTION KEY order_id;

\c regress

CREATE TABLE orders(id INT PRIMARY KEY);

CREATE TABLE delivery(id INT PRIMARY KEY, order_id INT, FOREIGN KEY(order_id) REFERENCES orders(id));

INSERT INTO orders(id) VALUES (5);
INSERT INTO delivery(id,order_id) VALUES (10, 5);
SELECT * FROM delivery;
SELECT * FROM delivery JOIN orders ON order_id = id;
SELECT * FROM delivery JOIN orders ON delivery.order_id = orders.id;

DROP TABLE orders CASCADE;
DROP TABLE delivery;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;