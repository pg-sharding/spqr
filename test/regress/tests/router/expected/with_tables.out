\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
         add distribution         
----------------------------------
 created distribution with id ds1
(1 row)

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
         add key range          
--------------------------------
 created key range with bound 1
(1 row)

CREATE KEY RANGE krid2 FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;
          add key range           
----------------------------------
 created key range with bound 101
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION orders DISTRIBUTION KEY id;
                 attach table                 
----------------------------------------------
 attached relation orders to distribution ds1
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION delivery DISTRIBUTION KEY order_id;
                  attach table                  
------------------------------------------------
 attached relation delivery to distribution ds1
(1 row)

\c regress
CREATE TABLE orders(id INT PRIMARY KEY);
NOTICE: send query to shard(s) : sh1,sh2
CREATE TABLE delivery(id INT PRIMARY KEY, order_id INT, FOREIGN KEY(order_id) REFERENCES orders(id));
NOTICE: send query to shard(s) : sh1,sh2
INSERT INTO orders(id) VALUES (5);
NOTICE: send query to shard(s) : sh1
INSERT INTO delivery(id,order_id) VALUES (10, 5);
NOTICE: send query to shard(s) : sh1
SELECT * FROM delivery;
NOTICE: send query to shard(s) : sh1,sh2
 id | order_id 
----+----------
 10 |        5
(1 row)

SELECT * FROM delivery JOIN orders ON order_id = id;
NOTICE: send query to shard(s) : sh1,sh2
ERROR:  column reference "id" is ambiguous
LINE 1: SELECT * FROM delivery JOIN orders ON order_id = id;
                                                         ^
SELECT * FROM delivery JOIN orders ON delivery.order_id = orders.id;
NOTICE: send query to shard(s) : sh1,sh2
 id | order_id | id 
----+----------+----
 10 |        5 |  5
(1 row)

DROP TABLE orders CASCADE;
NOTICE: send query to shard(s) : sh1,sh2
DROP TABLE delivery;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution   
-----------------------
 drop distribution ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

